### PR TITLE
Automatically convert non-flows into flows

### PR DESCRIPTION
**Pull request by jakekaplan**
This PR updates `load_flow_from_entrypoint` to automatically wrap any Python function in a `Flow` object if it is not already, removing the requirement that a user explicitly decorate their function with `@flow` if they want to deploy it. By doing so, we lower the barrier for using Prefect to remotely deploy and run code—allowing quick adoption without forcing users to modify existing codebases as existing scripts become immediately deployable/schedulable.

As far I can tell there is no technical reason why we can't do this and I think it is a very big value add. I would be happy only include this functionality behind an experimental flag if that is preferred.

### CODE REVIEW

**Comment by desertaxle** [comment]
My initial reaction is that this is cool and should be its own utility. That way we can keep the behavior of the current utility consistent and sidestep any edge cases that might arise if we get a `ScriptError` while loading a function that isn't already decorated with `@flow`.

**Comment by jakekaplan** [comment]
> My initial reaction is that this is cool and should be its own utility. That way we can keep the behavior of the current utility consistent and sidestep any edge cases that might arise if we get a `ScriptError` while loading a function that isn't already decorated with `@flow`.

That sounds good to me! I would still need modify the tests but I made a tweak. Is this what you had in mind? Or should I be doing this a different way/in a different place?

**Review by zzstoatzz — APPROVED** [review]
cool!

File: src/prefect/flows.py [hunk 1]
```diff
@@ -2080,6 +2080,29 @@ def load_flow_from_entrypoint(
...
+        # split by the last colon once to handle Windows paths with drive letters i.e C:\path\to\file.py:do_stuff
+        path, func_name = entrypoint.rsplit(":", maxsplit=1)
+    else:
+        path, func_name = entrypoint.rsplit(".", maxsplit=1)
+    try:
+        func = import_object(entrypoint)  # pyright: ignore[reportRedeclaration]
+    except AttributeError as exc:
+        raise MissingFlowError(
+            f"Function with name {func_name!r} not found in {path!r}. "
+        ) from exc
```
Comment (desertaxle) [inline src/prefect/flows.py:2098]: We probably want a different exception type in this function. It doesn't need to be a custom exception though and a `RuntimeError` should work.

File: tests/test_flows.py [hunk 2]
```diff
@@ -2695,6 +2696,38 @@ def dog():
             load_flow_from_entrypoint(f"{fpath}:dog", use_placeholder_flow=False)
 
 
+class TestLoadFunctionAndConvertToFlow:
```
Comment (desertaxle) [inline tests/test_flows.py:2699]: Can you add a test covering when the function isn't found?

File: src/prefect/flows.py (hunk 1 above)
Comment (jakekaplan) [reply src/prefect/flows.py:2098]: Got it will update

File: tests/test_flows.py (hunk 2 above)
Comment (jakekaplan) [reply tests/test_flows.py:2699]: Yes can do!
